Python/instrumentation.c

```diff
@@ -1741,19 +1763,22 @@ _PyMonitoring_SetLocalEvents(PyCodeObject *code, int tool_id, _PyMonitoringEvent
 {
     assert(0 <= tool_id && tool_id < PY_MONITORING_TOOL_IDS);
     PyInterpreterState *interp = _PyInterpreterState_GET();
-    assert(events < (1 << _PY_MONITORING_UNGROUPED_EVENTS));
+    assert(events < (1 << _PY_MONITORING_LOCAL_EVENTS));
+    if (code->_co_firsttraceable >= Py_SIZE(code)) {
+        PyErr_Format(PyExc_SystemError, "cannot instrument shim code object", code->co_name);
```

Comment:
If `code->_co_firsttraceable >= Py_SIZE(code)` then the error is that the code object is visible to the user at all.

Hypothetically it could be through `sys.get_referrers(None)`, but anyone who does and then tries to instrument the code object should not be surprised to get strange errors.

Reply:
Except it isn't tracked, so you can't even find it that way.
So it should be impossible, so I think a SystemError is best